chapters/iterative-search.qmd

```diff
@@ -0,0 +1,1027 @@
+
+How should the candidates be perturbed from iteration to iteration? When the tuning parameters are all numeric, we can create a random distance and angle from the current values. A similar process can be used for integers by "flooring" them to the nearest whole number. For qualitative tuning parameters, a random subset of parameters is chosen to change to a different value chosen at random. The amount of change should be large enough to search the parameters space and potentially get out of a local optimum.
+
+It is also important to conduct computations on the parameters in their potentially transformed space. This ensures that the entire range of the parameter values is treated equally. For example, we’ve been working with the values on their log (base 2) scale for the SVM cost parameter. When perturbing the values, we make sure to change them in the log space. This is true of all of the other search methods described in this chapter. 
+
+To illustrate, a very small initial space-filling design with three candidates was generated and evaluated. The results are in @tbl-svm-initial. To start the SA search^[These values will be also used as the initial substrate for Bayesian optimization.], we will start with the candidate with the smallest RMSE and proceed for 50 iterations without a rule for early stopping. A restart to the last known best results was enforced after eight suboptimal iterations. 
+
+:::: {.columns}
+
+::: {.column width="25%"}
+:::
+
+::: {.column width="50%"}
+
+::: {#tbl-svm-initial}
+
+```{r}
+#| label: tbl-svm-initial
+#| echo: false
+sa_history %>% 
+  filter(.iter == 0) %>% 
+  arrange(mean) %>% 
+  select(RMSE, log_cost, log_scale) %>% 
+  kable(
+    digits = 1,
+    col.names  = c("RMSE (%)", "Cost (log-2)", "Scale (log-10)")
+  ) %>% 
+  kable_styling(bootstrap_options = "responsive", full_width = FALSE)
+
+load(url("https://raw.githubusercontent.com/aml4td/website/main/RData/two_param_iter_initial.RData"))
+```
+
+The initial set of candidates used for iterative search with the toy example from @fig-svm-grid. One candidate does poorly while the other two have relatively similar results. 
+
+:::
+
+:::
+
+::: {.column width="25%"}
+:::
+
+:::: 
+ 
+@fig-sa-example contains an animation of the results of the SA search. In the figure, the initial points are represented by open circles, and a grey diagonal line shows the ridge of values exists that corresponds to the best RMSE results. 
+
+:::: {.columns}
+
+::: {.column width="15%"}
+:::
+
+::: {.column width="70%"}
+
+::: {#fig-sa-example}
+
+::: {.figure-content}
+
+```{shinylive-r}
```

Comment:
Once these are merged, I will move these apps to GitHub in individual files so that this file is much simplier.